napari/layers/shapes/_shapes_mouse_bindings.py

```diff
@@ -101,8 +102,27 @@ def select(layer: Shapes, event: MouseEvent) -> Generator[None, None, None]:
         # Drag any selected shapes
         if len(layer.selected_data) == 0:
             _drag_selection_box(layer, coordinates)
-        else:
-            _move_active_element_under_cursor(layer, coordinates)
+        elif not is_moving:
+            if vertex_under_cursor is not None:
+                vertex_indices = ((vertex_under_cursor,),)
+            else:
+                vertex_indices = tuple(
+                    tuple(
+                        vertex_index
+                        for vertex_index, coord in enumerate(layer.data[i])
+                    )
+                    for i in layer.selected_data
+                )
+            layer.events.data(
+                value=layer.data,
+                action=ActionType.CHANGING,
+                data_indices=tuple(
+                    layer.selected_data,
+                ),
+                vertex_indices=vertex_indices,
+            )
```

Comment:
whis should be in Shapes inplementation

Reply:
you mean similar to Points moving all the events the the actual `Layer`? I agree, but I do think we agreed to it that, that would be a follow up PR. If not I can make it part of this pr.

Reply:
Would you propose to move `_move_active_element_under_cursor` to the `Shapes` layer itself for this?

Reply:
I do not understand. You have added event emission in the `select` function, not in `_move_active_element_under_cursor`. I see that `_move_active_element_under_cursor` also contains event emission, but this event is not added in this PR. It is modified, but as you spot, the `_move_active_element_under_cursor` will be easy to move to `Shapes` class. When this event added here cannot be easily moved.